DataStructureTwo/DataStructureTwo/Counseling.java

```diff
@@ -0,0 +1,81 @@
+package DataStructureTwo;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Logger;
+/**
+ * 
+ * @author Vinod Patel
+ *
+ */
+public class Counseling {
+
+	private static Logger logger = Logger.getLogger(Counseling.class.getName());
+	
+	/**
+	 * Counseling starts here
+	 * @param candidatesQueue
+	 * @param collegeList
+	 * @param input
+	 * @return list of candidate status 
+	 */
+	public List<String> startCounselling(Queue<Candidates> candidatesQueue,List<College> collegeList, int[] input){
+		int inputCounter = 0 ;
+		int collegeChoice;
+		int totalStudents = candidatesQueue.getSize();
+		List<Candidates> candidateList = new ArrayList<Candidates>();
+		Candidates[] candidatesArray = new Candidates[totalStudents];
+		candidatesArray = candidatesQueue.toArray(candidatesArray);
+		/**
+		 * sorting candidates on the bases of marks.
+		 */
+		Arrays.sort(candidatesArray);
+		
+		//making queue empty
+		candidatesQueue.makeEmpty();
+		
+		//reassigning queue in sorted form
+		for(int i = 0 ; i < totalStudents ; i++ ){
+			Candidates item = new Candidates(candidatesArray[i].getName(),candidatesArray[i].getMarks());
+			candidatesQueue.enqueue(item);
+		}
+		
+		for(int i = 0 ; i < totalStudents ; i++ ){
+			Candidates candidate = candidatesQueue.dequeue();
+			candidateList.add(candidate);
+			logger.info("Call for Student:"+ candidate.getName() +" Remaining Students = "+candidatesQueue.getSize());
+			boolean flag = false;
+			int j ;
+			
+			for( j=0 ; j < collegeList.size() ; j++ ){
+				if(collegeList.get(j).isSeatAvailable()) {
```

Comment:
store collegeList.get(j) into a local variable then use that local variable. I think you need to learn the proper use of local varaible.